tests/test_engage.py

```diff
@@ -142,3 +140,38 @@ def test_engage(git_repo, test_args, capsys):
     setattr(test_args, "input_data", "data")
     with pytest.raises(AssertionError):
         engage(test_args)
+
+def test_csv_with_ext(git_repo, test_args, capsys):
+    setattr(test_args, "csv", "catalogue_res.csv")
+
+    engage(test_args)
+    results_path = os.path.join(git_repo, "results")
+    setattr(test_args, "output_data", results_path)
+    setattr(test_args, 'command', 'disengage')
+    disengage(test_args)
+    output_file = glob.glob("catalogue_results/*.csv")
+    print(output_file)
+    assert len(output_file) == 1
+
+    # clean up: delete files created in CWD
+    os.remove(output_file[0])
+    os.rmdir("catalogue_results")
+
+def test_csv_without_ext(git_repo, test_args, capsys):
+    setattr(test_args, "csv", "catalog2ue_res")
+    engage(test_args)
+    results_path = os.path.join(git_repo, "results")
+    setattr(test_args, "output_data", results_path)
+    setattr(test_args, 'command', 'disengage')
+    disengage(test_args)
+    output_filecsv = glob.glob("catalogue_results/*.csv")
+    output_filejson = glob.glob("catalogue_results/*.json")
+    print(output_filecsv)
+
+    print(output_filejson)
+    assert len(output_filecsv) == 0
+    assert len(output_filejson) == 1
```

Comment:
This test doesn't work for me - it seems that no json file is created

Reply:
hmm that's odd. What did you get as your result from printing?
I got something like `[]` for `output_filecsv` and `['catalogue_results/202000827-163438.json']` for the `output_filejson`.